// flx-ui-datatable/esm5/lib/flx-ui-datatable-service.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Injectable } from '@angular/core';
import { FlxUiDatatableService } from './flx-ui-datatable.service';
import { BehaviorSubject } from 'rxjs';
var FlxUiDataTable = /** @class */ (function () {
    function FlxUiDataTable(service) {
        var _this = this;
        this.service = service;
        this.behavior = new BehaviorSubject([]);
        this.flxDatatableData = this.behavior.asObservable();
        this.service.flxData.subscribe(function (resp) {
            _this.ChangeData(resp);
        });
    }
    /**
     *
     * @param data Change table data with new data
     */
    /**
     *
     * @param {?} data Change table data with new data
     * @return {?}
     */
    FlxUiDataTable.prototype.ChangeData = /**
     *
     * @param {?} data Change table data with new data
     * @return {?}
     */
    function (data) {
        this.behavior.next(data);
    };
    /**
     * Reload data from api: void
     * @return {?}
     */
    FlxUiDataTable.prototype.reloadData = /**
     * Reload data from api: void
     * @return {?}
     */
    function () {
        this.service.loadFlxDataTableData(this.service.getDataUrl());
    };
    /**
     * @return {?}
     */
    FlxUiDataTable.prototype.abortRequest = /**
     * @return {?}
     */
    function () {
        this.service.cancelLoading();
    };
    FlxUiDataTable.decorators = [
        { type: Injectable },
    ];
    /** @nocollapse */
    FlxUiDataTable.ctorParameters = function () { return [
        { type: FlxUiDatatableService }
    ]; };
    return FlxUiDataTable;
}());
export { FlxUiDataTable };
function FlxUiDataTable_tsickle_Closure_declarations() {
    /** @type {?} */
    FlxUiDataTable.prototype.behavior;
    /** @type {?} */
    FlxUiDataTable.prototype.flxDatatableData;
    /** @type {?} */
    FlxUiDataTable.prototype.service;
}
